infrastructure/constructs/api.py

```diff
@@ -37,7 +46,7 @@ def __init__(  # pylint: disable=too-many-arguments,too-many-locals
             self,
             "api-users-role",
             role_name=Resource.API_USERS_ROLE_NAME.resource_name,
-            assumed_by=principal,  # type: ignore[arg-type]
```

Comment:
`principal` is presumably still the same type, so I don't understand why this change is in this PR. Was it necessary to pass linting?